### PR TITLE
Fix the AI acting the same at game restart.

### LINKED ISSUE #13256: AI acts quite predeterminated.
Notsure what's causing this though.

Observe a full AI match and restart the game from the options menu. The game then ends up recreating the same situations.

### PR DESCRIPTION
**Pull request by GraionDilach**
Fixes #13256.

### CODE REVIEW

File: OpenRA.Mods.Common/AI/HackyAI.cs [hunk 1]
```diff
@@ -342,7 +342,7 @@ public void Activate(Player p)
 			foreach (var defense in Info.DefenseQueues)
 				builders.Add(new BaseBuilder(this, defense, p, playerPower, playerResource));
 
-			Random = new MersenneTwister((int)p.PlayerActor.ActorID);
+			Random = new MersenneTwister(Game.CosmeticRandom.Next());
```
Comment (rob-v) [inline OpenRA.Mods.Common/AI/HackyAI.cs]: This works, but I would use only Random = new MersenneTwister(); so without dependency on Game.CosmeticRandom.

Comment (pchote) [inline OpenRA.Mods.Common/AI/HackyAI.cs]: The problem there is that multiple bots would be seeded with the same value if they are constructed within the same millisecond.

Comment (rob-v) [inline OpenRA.Mods.Common/AI/HackyAI.cs]: Ok, I assumed in practice that can't happen.

**Comment by rob-v** [comment]
👍

**Review by chrisforbes — APPROVED** [review]

**Review by pchote — APPROVED** [review]

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/3d1748b3a60db33012b88a1dd83bdb4df9311867

**Pull request merged**